meson.build

```diff
@@ -163,11 +163,40 @@ endif
 xsf = subproject('xsf')
 xsf_dep = xsf.get_variable('xsf_dep')
 
-boost_math_dep = dependency(
-  'boost',
-  version : '1.88.0',
-  fallback : ['boost_math', 'boost_math_dep'],
-)
+use_system_libraries = get_option('use-system-libraries')
+all_system_libraries = false
+auto_system_libraries = false
+
+if use_system_libraries.contains('none')
+  use_system_libraries = ['none']
+elif use_system_libraries.contains('all')
+  all_system_libraries = true
+elif use_system_libraries.contains('auto')
+  auto_system_libraries = true 
+endif
+
+if all_system_libraries or use_system_libraries.contains('boost.math')
+  boost_math_dep = dependency('boost', version : '1.88.0')
+elif auto_system_libraries
+  boost_math_dep = dependency(
+    'boost', version : '1.88.0',
+    fallback : ['boost_math', 'boost_math_dep']
+  )
+else
+  boost_math = subproject('boost_math', version : '1.88.0')
+  boost_math_dep = boost_math.get_variable('boost_math_dep')
```

Comment:
All of this would be handled using
```meson
   boost_math_dep = dependency(
    'boost', version : '1.88.0',
    fallback : ['boost_math', 'boost_math_dep']
  )
```